File: typescript-connector/src/models/Brand.ts
// Class from DFC Business Ontology: #Brand
import { SemanticObject } from "../core/SemanticObject.js";
import { WhatSubject, type WhatSubjectParams } from "./WhatSubject.js";

export interface BrandParams extends WhatSubjectParams {
  brandOf?: string;
  ownedBy?: string;
}
export class Brand extends WhatSubject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:Brand";
  }

  brandOf?: string;
  ownedBy?: string;

  constructor(
    semanticId: string,
    params?: BrandParams,
  ) {
    super(semanticId, params);
    this.brandOf = params?.brandOf;
    this.ownedBy = params?.ownedBy;
    this.semanticType = Brand.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:Brand:brand_of", () => this.brandOf);
    this.registerSemanticProperty("dfc-b:Brand:owned_by", () => this.ownedBy);
  }
  static {
    SemanticObject.typeRegistry.set(Brand.SEMANTIC_TYPE, Brand);
  }
}

File: ruby-gem/lib/models/brand.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #Brand
require_relative '../core/semantic_object'
require_relative 'what_subject'


module DfcLinkmlConnector
  module Models
    class Brand < WhatSubject
      SEMANTIC_TYPE = "dfc-b:Brand".freeze

      # @return [String]
      attr_accessor :brand_of

      # @return [String]
      attr_accessor :owned_by

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, brandOf: nil, ownedBy: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, brandOf: nil, ownedBy: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)
        @brand_of = brandOf
        @owned_by = ownedBy
        self.semanticType = "dfc-b:Brand"
        registerSemanticProperty("dfc-b:brandOf", &method("brand_of")).valueSetter = method("brand_of=")
        registerSemanticProperty("dfc-b:ownedBy", &method("owned_by")).valueSetter = method("owned_by=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
